src/FreeformIdeas.js

```diff
@@ -33,26 +47,36 @@ const FreeForm = ({ hideSourceOnDrag }) => {
             },
         }));
     };
+    const editIdea = useCallback((title) => {
+        if(title !=null){
+            setIdeas(update(ideas, {
+                [storedID]:{
+                    $merge:{title},
+                },
+            }))
+        };
+    }, [ideas, storedID]);
     const addIdea = useCallback((title) => {
         const id = uuidv4()
-        const top = 200;
-        const left = 200;
+        const left = coords[0];
+        const top = coords[1];
         setIdeas(update(ideas, {[id]: {$set: {top,left,title}}}));
-    }, [ideas]);
+    }, [coords, ideas]);
     const handleClick = useCallback((e) => {
         if(e.target.className !== "FreeformMap") return;
         e.preventDefault();
+        setCoords([e.nativeEvent.layerX,e.nativeEvent.layerY]);
         setTimeout(showPopup,300);
     }, [showPopup]);
 
     return (
-        <div ref={drop} onDoubleClick={handleClick} className='FreeformMap'>
-            <Popup show={popupShow} onClose={hidePopup} onSubmit={addIdea} />
+        <div ref={drop} onDoubleClick={handleClick} className="FreeformMap">
+            <Popup show={popupShow} onClose={hidePopup} onSubmit={formType ? addIdea : editIdea} formType={formType}/>
             {
                 Object.keys(ideas).map((key) => {
                     const { left, top, title } = ideas[key];
                     return (
-                        <Idea key={key} id = {key} left = {left} top = {top} hideSourceOnDrag={hideSourceOnDrag}>
+                        <Idea key={key} id = {key} left = {left} top = {top} hideSourceOnDrag={hideSourceOnDrag} onEdit={showEdit} >
```

Comment:
```suggestion
                        <Idea key={key} id = {key} left = {left} top = {top} onEdit={showEdit} >
```

Reply:
for this, instead of just resolving the conversation, should I be committing the suggestion?